core/src/storage/impls/multi_version_merkle_patricia_trie/merkle_patricia_trie/mod.rs

```diff
@@ -17,6 +17,49 @@ pub(self) mod walk;
 #[cfg(test)]
 mod tests;
 
+/// The purpose of this trait is to create a new value of a passed object,
+/// when the passed object is the value, simply move the value;
+/// when the passed object is the reference, create the new value by clone.
+/// Other extension is possible.
+///
+/// The trait is used by ChildrenTable and slab.
+pub trait WrappedCreateFrom<FromType, ToType> {
```

Comment:
Maybe it's better to make `ToType` an associated type?

Reply:
Not exactly. Associate type means that you can't have another ToType2 for the same FromType.